### DISCUSSION TITLE
Missing line feed(s) in PDF in text block

### DISCUSSION THREAD

**Discussion by darkandrew7** [pymupdf/PyMuPDF#1285] [Upstream Bugs] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
When placing redact annotations on the attached document, it seems there's a LF discarded during `apply_redactions`. This causes text that should span multiple lines to be written on one line.

## To Reproduce (mandatory)
Place a redaction annotation anywhere on the document and finalize with `Page.apply_redactions()`. 
```python
# give me a few minutes to verify the code 
```

## Expected behavior (optional)
There shouldn't be overlapping text. 

## Screenshots (optional)
![incorrect](https://user-images.githubusercontent.com/29284192/134701426-eafedec9-8f29-4ce4-bb48-28994a06d4c0.png)
![correct](https://user-images.githubusercontent.com/29284192/134701427-8c42aadc-0acb-497e-8b86-3a6e3b25f2b5.png)

## Your configuration (mandatory)
 - Operating system: debian 10
 - Python version: 3.8
 - PyMuPDF version: 1.18.16 from `pip`

Problem document:
[textbox_linefeed_glitch.pdf](https://github.com/pymupdf/PyMuPDF/files/7226844/textbox_linefeed_glitch.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
This is an upstream bug (MuPDF) unfortunately.
They have fixed it in their development version, but that is unreleased yet.

**Comment by darkandrew7** [comment]
That's what I assumed; I thought I'd report it anyhow. Thank you so much for your help!

**Comment by JorjMcKie** [comment]
Because there is no wy for me to help you, I am putting this issue on the pile in Discussions ☹️
